mesonbuild/interpreter.py

```diff
@@ -1615,10 +1616,15 @@ def has_multi_arguments_method(self, args, kwargs):
     @permittedKwargs({})
     def get_supported_arguments_method(self, args, kwargs):
         args = mesonlib.stringlistify(args)
-        supported_args = []
-        for arg in args:
-            if self.has_argument_method(arg, kwargs):
-                supported_args.append(arg)
+        with conc.ThreadPoolExecutor() as executor:
+            futures = []
+            for arg in args:
+                f = executor.submit(self.has_argument_method(arg, kwargs))
```

Comment:
How can this even work? You just submitted the result of the check to the thread pool, not the check itself.

Reply:
It submit them all, then call result() on each. That should block until they complete, no?

Reply:
Forget me, I see what you mean... Why does it make a big difference in time... Maybe I'm just to tired... :P

Reply:
Voilà fixed, the gain is much less impressive, but still 10s less. Should not code during the evening after FOSDEM, too tired :P